src/Analyzer/Pass/Expression/FunctionCall/FunctionStringFormater.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace PHPSA\Analyzer\Pass\Expression\FunctionCall;
+
+use PHPSA\Context;
+use PhpParser\Node;
+use PhpParser\Node\Stmt;
+use PHPSA\Analyzer\Pass;
+use PhpParser\Node\Expr\Array_;
+use PhpParser\Node\Expr\FuncCall;
+use PhpParser\Node\Scalar\String_;
+use PHPSA\Analyzer\Helper\DefaultMetadataPassTrait;
+
+class FunctionStringFormater extends AbstractFunctionCallAnalyzer
+{
+    use DefaultMetadataPassTrait;
+
+    const DESCRIPTION = 'Format string has same number of placeholders as parameters are passed into and forbid invalid type formats.';
+
+    /**
+     * @var array functions
+     */
+    protected static $functions = [
+        'printf' => 'printf',
+        'sprintf' => 'sprintf',
+        'vprintf' => 'vprintf',
+        'vsprintf' => 'vsprintf'
+    ];
+    /**
+     * Placeholders for type format
+     * @var array
+     */
+    protected $placeholders = [];
+
+    /**
+     * @param FuncCall $funcCall
+     * @param Context $context
+     * @return bool
+     */
+    public function pass(FuncCall $funcCall, Context $context)
+    {
+        $functionName = $this->resolveFunctionName($funcCall, $context);
+        if ($functionName && isset(self::$functions[$functionName])) {
+            if ($funcCall->args) {
+                $args = $funcCall->args;
+
+                if (! ($args[0]->value instanceof String_)) {
```

Comment:
You should compile $args[$i]->value because it can be varialbe with string,

```
$a = "1";
sprinf("%s", $a);
```

Reply:
Fixed in https://github.com/ovr/phpsa/commit/354ecf44cee1a0dcade75ae078b5f3be9512227b